qualtran/drawing/flame_graph.py

```diff
@@ -0,0 +1,124 @@
+            ret += f'[{getattr(bloq, field.name)}]'
+    return ret
+
+
+@functools.lru_cache(maxsize=None)
+def _t_counts_for_bloq(bloq: Bloq, graph: nx.DiGraph) -> int:
+    sigma = _compute_sigma(bloq, graph)
+    return t_counts_from_sigma(sigma)
+
+
+def _keep_if_small(bloq: Bloq) -> bool:
+    from qualtran.bloqs.and_bloq import And
+    from qualtran.bloqs.basic_gates import CSwap, Toffoli
+
+    if isinstance(bloq, (And, Toffoli, CSwap)):
+        return True
+
+
+def _populate_flame_graph_data(
+    bloq: Bloq, graph: nx.DiGraph, graph_t: nx.DiGraph, prefix: List[str]
+) -> List[str]:
+    callees = [x for x in list(graph.successors(bloq)) if _t_counts_for_bloq(x, graph_t) > 0]
+    total_t_counts = _t_counts_for_bloq(bloq, graph_t)
+    prefix.append(_pretty_name(bloq) + f'(T:{total_t_counts})')
+    data = []
+    if len(callees) == 0 or (len(callees) == 1 and callees[0] in [TGate(), TGate(is_adjoint=True)]):
+        data += [';'.join(prefix) + '\t' + str(total_t_counts)]
+    else:
+        succ_t_counts = 0
+        for succ in callees:
+            curr_data = _populate_flame_graph_data(succ, graph, graph_t, prefix)
+            succ_t_counts += (
+                _t_counts_for_bloq(succ, graph_t) * graph.get_edge_data(bloq, succ)['n']
+            )
+            data += curr_data * graph.get_edge_data(bloq, succ)['n']
+        # assert total_t_counts == succ_t_counts, f'{bloq=}, {total_t_counts=}, {succ_t_counts=}'
+    prefix.pop()
+    return data
+
+
+def get_flame_graph_data(
+    *bloqs: Bloq,
+    file_path: Union[None, pathlib.Path, str] = None,
+    keep: Optional[Sequence['Bloq']] = _keep_if_small,
+    **kwargs,
+) -> str:
+    data = []
+    for bloq in bloqs:
+        call_graph, _ = bloq.call_graph(keep=keep, **kwargs)
+        call_graph_t_counts, _ = bloq.call_graph()
+        data += _populate_flame_graph_data(bloq, call_graph, call_graph_t_counts, prefix=[])
+    if file_path:
+        with open(file_path, 'w') as f:
+            f.write('\n'.join(data))
+    else:
+        return '\n'.join(data)
+
+
+def show_flame_graph(
+    *bloqs: Bloq, file_path: Union[None, pathlib.Path, str] = None, **kwargs
+) -> None:
+    data = get_flame_graph_data(*bloqs, **kwargs)
+    import pathlib
+    import subprocess
+
+    pipe = subprocess.Popen(
+        [pathlib.Path(__file__).resolve().parent / "flamegraph.pl --countname='TCounts' "],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        shell=True,
+    )
+    pipe.stdin.write(data)
+    svg_data = pipe.communicate(input=data)[0]
+    pipe.stdin.close()
+    if file_path:
+        with open(file_path, 'w') as f:
+            f.write(svg_data)
+    else:
+
+        @attrs.frozen
+        class _InteractiveSVG:
+            svg_content: str
+
+            def _repr_html_(self):
+                return self.svg_content
+
+        IPython.display.display(_InteractiveSVG(svg_data))
```

Comment:
explain why this is needed over `IPython.display.SVG`

Reply:
Removed. The idea was that when you save `svg_data` to a file and open that file as an svg, the file is interactive so you can zoom in  / out. I was hoping to achieve the same effect in jupyter notebook but this was a failed attempt at it. For now, jupyter notebook visualization would be static.